### ISSUE TITLE
2.80 linux version error:  NameError: name 'location' is not defined

### ISSUE THREAD

**Issue by sorinv** [JacquesLucke/animation_nodes#1168]
We found this issue on linux with several clients projects (can't share them unfortunately). It seems to affect only the Linux build, Windows and Mac do not have it.

On Linux, we are rendering from command line, but I tried command line on Windows as well, so I don't think this is what causes the issue.

Version affected: 2.1.4
```
Traceback (most recent call last):
  File "/blender/2.80/2.80/scripts/addons/animation_nodes/execution/main_execution_unit.py", line 38, in executeUnit
    exec(self.executeCodeObject, self.executionData, self.executionData)
  File "execution: 'Cross Cloud Atmosphere'", line 18, in <module>
NameError: name 'location' is not defined
```

**Comment by OmarSquircleArt** [comment]
Do you know which node causes this? Can you provide the execution code by going to the *Developer Panel* and pressing *Print*? The code should be printed to the terminal.

**Comment by sorinv** [comment]
This is from a different project that shows the same error. I can get another print of the code tomorrow from the project with the error above.

```
  1.  import bpy, sys
  2.  import itertools
  3.  from time import perf_counter as getCurrentTime
  4.  from mathutils import Vector, Matrix, Quaternion, Euler
  5.  AN = animation_nodes = sys.modules.get('animation_nodes')
  6.  from animation_nodes.data_structures import *
  7.  from animation_nodes import algorithms
  8.  _node_execution_times = animation_nodes.execution.measurements.getMeasurementsDict()
  9.  nodes = bpy.data.node_groups['Cross Cloud Atmosphere'].nodes
 10.  _ln8pbmoxbyd76ni = nodes['Time Info']
 11.  _192st9b53mun9lr = nodes['Object Instancer']
 12.  _j9ep14a8e4xnkyr = nodes['Vector Wiggle']
 13.  _9i7dyv3d2mz7zj3 = nodes['Object Transforms Output']
 14.  _scene_ln80 = _ln8pbmoxbyd76ni.inputs[0].getValue()
 15.  _instancesAmount_1920 = _192st9b53mun9lr.inputs[0].getValue()
 16.  _sourceObject_1921 = _192st9b53mun9lr.inputs[1].getValue()
 17.  _scenes_1922 = _192st9b53mun9lr.inputs[2].getValue()
 18.  _seed_j9e0 = _j9ep14a8e4xnkyr.inputs[0].getValue()
 19.  _count_j9e1 = _j9ep14a8e4xnkyr.inputs[1].getValue()
 20.  _speed_j9e3 = _j9ep14a8e4xnkyr.inputs[3].getValue()
 21.  _amplitude_j9e4 = _j9ep14a8e4xnkyr.inputs[4].getValue()
 22.  _octaves_j9e5 = _j9ep14a8e4xnkyr.inputs[5].getValue()
 23.  _persistance_j9e6 = _j9ep14a8e4xnkyr.inputs[6].getValue()
 24.  _rotation_9i72 = _9i7dyv3d2mz7zj3.inputs[2].getValue()
 25.  _scale_9i73 = _9i7dyv3d2mz7zj3.inputs[3].getValue()


  1.
  2.  # Node: 'Cross Cloud Atmosphere' - 'Time Info'
  3.  if _scene_ln80 is not None:
  4.      _frame_ln81 = _scene_ln80.frame_current_final
  5.  else:
  6.      _frame_ln81 = _startFrame_ln82 = _endFrame_ln83 = _frameRate_ln84 = 0
  7.
  8.  # Node: 'Cross Cloud Atmosphere' - 'Object Instancer'
  9.  _scenes = set(_scenes_1922)
 10.  _objects_1923 = _192st9b53mun9lr.getInstances_WithSource(_instancesAmount_1920, _sourceObject_1921, _scenes)
 11.
 12.  # Node: 'Cross Cloud Atmosphere' - 'Vector Wiggle'
 13.  _seed = _seed_j9e0 * 23452 + _j9ep14a8e4xnkyr.nodeSeed * 643523
 14.  _vectors_j9e7 = algorithms.perlin_noise.wiggleVectorList(max(0, _count_j9e1), _seed + _frame_ln81 * _speed_j9e3 / 20, _amplitude_j9e4, _octaves_
j9e5, _persistance_j9e6)
 15.
 16.  # Node: 'Cross Cloud Atmosphere' - 'Object Transforms Output'
 17.  iterations = len(_objects_1923)
 18.  objects_iter = _objects_1923
 19.  if len(_vectors_j9e7) >= iterations:
 20.      locations_iter = _vectors_j9e7
 21.  elif len(_vectors_j9e7) == 0:
 22.      locations_iter = itertools.cycle([_9i7dyv3d2mz7zj3.inputs[1].baseType.getDefaultValue()])
 23.  elif len(_vectors_j9e7) < iterations:
 24.      locations_iter = itertools.cycle(_vectors_j9e7)
 25.  vectorizeIterator = zip(objects_iter, locations_iter)
 26.  for _base_object, _base_location in vectorizeIterator:
 27.      if _base_object is not None:
 28.          _base_object.location = _base_location
 29.      pass
```

**Comment by OmarSquircleArt** [comment]
It would be nice if you can give me the error for this particular project, since the line numbers are different.

**Comment by sorinv** [comment]
I have updated the error message in the first message to the one printed by the code about. Doesn't make too much sense to me, as the line number reported doesn't seem to have anything to do with  location operations.

**Comment by OmarSquircleArt** [comment]
@sorinv I see. Please note that the *Print* operator will only print the execution code of the selected execution unit. An execution unit is a single connected group of nodes, so if you have multiple sets of nodes that are not connected to each other, try selecting each one individually and using the *Print* operator.

**Comment by sorinv** [comment]
It's from the correct group, named "Cross Cloud Atmosphere":
error: `File "execution: 'Cross Cloud Atmosphere'", line 18, in <module>`
log: ` # Node: 'Cross Cloud Atmosphere' - 'Object Transforms Output'`

**Comment by OmarSquircleArt** [comment]
@sorinv Can you try to replicate the error using a minimal node tree?

**Comment by sorinv** [comment]
Fortunately it was not difficult to replicate. blend file attached. Here's the log from the linux run:
```
Traceback (most recent call last):
  File "/blender/2.80/2.80/scripts/addons/animation_nodes/execution/main_execution_unit.py", line 38, in executeUnit
    exec(self.executeCodeObject, self.executionData, self.executionData)
  File "execution: 'NodeTree'", line 18, in <module>
NameError: name 'location' is not defined
```
[bug-aug-19.zip](https://github.com/JacquesLucke/animation_nodes/files/3488760/bug-aug-19.zip)

Let me know if there's anything else I can help with on this.

Thanks,
Sorin

**Comment by OmarSquircleArt** [comment]
@sorinv This file works fine for me. Which version and build are you using exactly? The one from graphicall?

**Comment by 3DSinghVFX** [comment]
@sorinv Your blend file is working on my system (Ubuntu 16.04) and I'm using the Linux build from graphicall https://blender.community/c/graphicall/Wdbbbc/.

**Comment by sorinv** [comment]
Exact same version form graphicall, just Ubuntu 18.04

**Comment by sorinv** [comment]
You may have to try rendering from command line. It may also be related to the fact that the server rendering this is headless, with no graphics card connected.

**Comment by OmarSquircleArt** [comment]
Ah I see. Good point. Will try that.

**Comment by OmarSquircleArt** [comment]
@sorinv I still can't replicate that, even while rendering in background mode from the terminal. 

@JacquesLucke Any idea?

**Comment by sorinv** [comment]
After looking some more into this, I managed to track down this error to the moment the animation nodes addon is enabled. 

This file enables the animation nodes addon:
```
$ cat /tmp/enable-an.py
import bpy

bpy.ops.preferences.addon_enable(module="animation_nodes")
```

This command line fails:
```
./blender -b /tmp/bug-aug-19.blend -P /tmp/enable-an.py -x 1 -F PNG -f 100
```

This command line works:
```
./blender -b -P /tmp/enable-an.py /tmp/bug-aug-19.blend -x 1 -F PNG -f 100
```

Previous AN build worked with the -P flag after the file too.

As you know, Blender executes the command line in order, so the version that works will enable animation nodes before the file is loaded, the second one will enable it after it is loaded.

Any chance to get it to work as it did before, even if AN is enabled after the file is loaded?

Thanks!

**Comment by OmarSquircleArt** [comment]
Thanks for the info @sorinv . I will look into this.

**Comment by OmarEmaraDev** [comment]
Finally managed to reproduce. Could be related to c47d4df.

**Comment by sorinv** [comment]
Great to hear you managed to reproduce it. 

I very much doubt  c47d4df fixed this. That's because we already use build 2.1.5 and it's still presenting the same problem.

**Comment by OmarEmaraDev** [comment]
I don't mean that `c47d4df` fixed the issue. Just saying that the issue seem to be around the vectorization code and this commit might be related.

**Comment by OmarEmaraDev** [comment]
Fixed in 6d97735.
